### PR TITLE
stlink flash fix bootloader path

### PR DESCRIPTION
**Pull request by benjinne**
fixes https://github.com/ExpressLRS/ExpressLRS/issues/2482
related to https://github.com/ExpressLRS/ExpressLRS/issues/2468

This uses args.fdir to set the bootloader path.
This works on my machine running ubuntu 22.04

### CODE REVIEW

File: src/python/binary_flash.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def upload_stm32_stlink(args, options: FirmwareOptions):
     app_start = flash_start + offset
 
     if options.bootloader is not None:
-        stlink.program_flash('bootloader/' + options.bootloader, flash_start, erase=True, verify=True, initialize_comms=True)
+        stlink.program_flash(args.fdir + '/bootloader/' + options.bootloader, flash_start, erase=True, verify=True, initialize_comms=True)
```
Comment (pkendall64) [inline src/python/binary_flash.py:65]: Should really calculate a local variable as `args.fdir` may be set to `None`, this is the pattern used in `binary_configurator.py:378`

```python
firmware_dir = '' if args.fdir is None else args.fdir + '/'
stlink.program_flash(firmware_dir + 'bootloader/' + options.bootloader, flash_start, erase=True, verify=True, initialize_comms=True)
```

Comment (benjinne) [reply src/python/binary_flash.py]: Thanks for the feedback. How does it look now?

Comment (pkendall64) [reply src/python/binary_flash.py]: Looks good. I'll give a test run this evening.

**Review by JyeSmith — APPROVED** [review]

**Comment by JyeSmith** [comment]
Does this need to target maintenance? @pkendall64

**Comment by pkendall64** [comment]
> Does this need to target maintenance? @pkendall64

Yes, once tested and approved I can take it und update the artifact repo files.

**Review by pkendall64 — APPROVED** [review]

**Pull request merged**